independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/options/BootstrapMavenOptions.java

```diff
@@ -101,16 +110,92 @@
     }
 
     public static BootstrapMavenOptions newInstance() {
-        return newInstance(getMavenCmdLine());
+        return newInstance(getMavenCmdLine(false).cmdLine);
     }
 
     public static BootstrapMavenOptions newInstance(String cmdLine) {
         return new BootstrapMavenOptions(parse(cmdLine));
     }
 
-    public static String getMavenCmdLine() {
+    public static class MavenCmdInfo {
+        public final String cmdLine;
+        public final List<String> userProps;
```

Comment:
I see why you did it this way but I am not sure we need another class to represent the command line, given that we already have BootstrapMavenOptions for that. As I mentioned in a comment above, I think BootstrapMavenOptions could have a method getOptionProperties(String) (similar to org.apache.commons.cli.CommandLine.getOptionProperties(String), in fact that should be the source of the value).

Reply:
My initial implementation was doing something very similar (it was returning a map of option name / value pairs). However, the only spot where this result is used is in DevMojo and it requires a list of formatted options, which is why I did it this way since I felt it was overkill to first produce a map just to transform it where it used.